documentation/specs/multithreaded-msbuild.md

```diff
@@ -0,0 +1,268 @@
+# Multithreaded MSBuild
+
+Currently, MSBuild supports parallel builds (a critical feature for a build system) by spawning worker processes. This made adoption easier because it didn't impose any requirements on tasks: they continue to own the whole process while they are executing. But it's a pretty strange design decision in the modern age, where we assume things are multithreaded and async.
```

Comment:
could you please give more context about the existing parallelisms here or point to the relevant documentation? 
e.g. does it parallel per project? or some tasks in the project can be executed in parallel?